app/presenters/api/v2/subheadings/subheading_presenter.rb

```diff
@@ -42,7 +42,7 @@ def chapter
         end
 
         def heading
-          @heading ||= ns_ancestors.find { |a| a.is_a? Heading }
+          @heading ||= ns_ancestors.reverse.find { |a| a.is_a? Heading }
```

Comment:
```suggestion
          @heading ||= ns_ancestors_dataset.declarable.find { |a| a.is_a? Heading }
```

Reply:
Its a micro optimisation for slightly large numbers of ancestors I think. No strong feeling re this and I actually feel like yours is more readable

Reply:
`ns_ancestors_dataset` isn't already loaded where as `ns_ancestors` is . In the precache scenario we load the chapter once, then feed the different headings and subheadings under it into the serializer - so this would introduce a N+1 query. I'll leave it as is I think because it'll be quicker to reverse the list then do a db query